centipede/symbol_table.h

```diff
@@ -126,18 +127,21 @@ class SymbolTable {
   void AddEntryInternal(std::string_view func, std::string_view file,
                         int line = -1, int col = -1);
 
-  std::string_view GetOrInsert(std::string_view str);
+  std::string GetOrInsert(std::string_view str);
 
   // Declaration order matters here, because we want `table_` to be deleted last
   // in order to avoid having dangling ptrs in `entries_`.
 
   // Holds the strings for files and function names of the stored symbols. This
   // avoids storing duplicate values. `node_hash_set` was chosen in order to
   // have pointer stability and not bother with storing strings in `unique_ptr`.
-  absl::node_hash_set<std::string> table_;
+  absl::node_hash_set<std::string> table_  ABSL_GUARDED_BY(mu_);
 
   // Holds the the symbol entries.
-  std::vector<Entry> entries_;
+  std::vector<Entry> entries_ ABSL_GUARDED_BY(mu_);
+
+  // A mutex to be claimed when accessing the table
+  absl::Mutex mu_;
```

Comment:
You want to define this as `mutable absl::Mutex mu_`, it will avoid the need for `const_cast` in const methods.